include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef HTTP2_SERVER_H
+#define HTTP2_SERVER_H
+
+#include <zephyr/posix/poll.h>
+
+#define MAX_CLIENTS 10
+
+struct http2_server_config {
+	int port;
+	int address_family;
+};
+
+struct http2_server_ctx {
+	int sockfd;
+	struct pollfd client_fds[MAX_CLIENTS];
+	int num_clients;
+};
+
+struct http2_frame {
```

Comment:
It might be better to define `struct http2_frame` elsewhere, e.g. in a private header. The telling part for me, is that `struct http2_frame` is not referenced anywhere in this header.